File: Android-Base/01-FourComponents/a-Activity/06_ActivityLife/activitytest/src/main/java/top/iqqcode/activitytest/BaseActivity.java
package top.iqqcode.activitytest;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;

public class BaseActivity extends AppCompatActivity {

    private static final String TAG = "TAG";
    static int count = 0;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Log.e(TAG, "[>> BaseActivity] ==> onCreate -- count=" + count);
    }

    @Override
    protected void onStart() {
        count++;
        Log.e(TAG, "[>> BaseActivity] ==> onStart -- count=" + count);
        super.onStart();
    }

    @Override
    protected void onStop() {
        count--;
        super.onStop();
        Log.e(TAG, "[>> BaseActivity] ==> onStop -- count=" + count);
    }
}

File: Kotlin/b-Exercises/01FindView/app_viewbinding/src/main/java/top/iqqcode/viewbinding/BaseActivity.kt
package top.iqqcode.viewbinding

import android.os.Bundle
import android.util.Log
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import top.iqqcode.viewbinding.databinding.ActivityBaseBinding

class BaseActivity : AppCompatActivity() {

    private lateinit var binding: ActivityBaseBinding
    private val fragment: FragmentDemo? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityBaseBinding.inflate(layoutInflater)
        setContentView(binding.root)

        val fragmentManager = supportFragmentManager
        fragmentManager.beginTransaction().add(R.id.fragmentContainer, FragmentDemo()).commit()
        val transaction = fragmentManager.beginTransaction()

        binding.btnReplace.setOnClickListener {
            Log.d("TAG", "onClick: btnReplace")
            transaction.replace(R.id.fragmentContainer, FragmentDemo2()).addToBackStack(null)
                .commit()
        }

        binding.btnRemove.setOnClickListener {
            if (fragment != null) transaction.remove(fragment).commit()
            Log.d("TAG", "onClick: btnRemove")
        }
    }


}